contracts/events/TopicEvent.sol

```diff
@@ -491,19 +458,43 @@ contract TopicEvent is ITopicEvent, Ownable, ReentrancyGuard {
     * @return The amount of BOT tokens won.
     */
     function calculateBotTokensWon() 
-        private
+        public
         view
+        inCollectionStatus()
         returns (uint256) 
     {
-        uint256 losingResultsTotal = 0;
-        for (uint8 i = 0; i < numOfResults; i++) {
-            if (i != finalResultIndex) {
-                losingResultsTotal = losingResultsTotal.add(balances[i].totalVoteBalance);
-            }
-        }
+        uint256 betBalance = balances[finalResultIndex].betBalances[msg.sender];
+        uint256 voteBalance = balances[finalResultIndex].voteBalances[msg.sender];
+        uint256 totalContribution = betBalance.add(voteBalance);
 
-        uint256 senderVoteBalance = balances[finalResultIndex].voteBalances[msg.sender];
-        uint256 winningResultTotal = balances[finalResultIndex].totalVoteBalance;
-        return senderVoteBalance.mul(losingResultsTotal).div(winningResultTotal).add(senderVoteBalance);
+        uint256 totalWinningBets = balances[finalResultIndex].totalBetBalance;
```

Comment:
what's the diff between `totalBetBalance ` and `totalVoteBalance `? Assuming `calculateBotTokensWon()` only accounts for one balance

Reply:
totalBetBalance = total QTUM deposited as bets
totalVoteBalance = total BOT deposited as votes

this is the calculation for QTUM won:
```
individualWinnerQtumBet * losersTotalQtumBet / winnersTotalQtumBet
```

This same calculation cannot work for calculating BOT tokens won, because of this case: 
- first round QTUM betting finishes and say result index is 0
- then a VotingOracle gets created
- losers from first round of betting bet on VotingOracle, but they do not hit the threshold
- so that means the winners who bet QTUM only do not have any BOT deposited

So I came up with the idea of calculating total contributions (both QTUM and BOT) to calculating the BOT won:
```
individualWinnerTotalContribution * losersTotalBotBet / winnersTotalContribution
```

does that make sense?